package.json

```diff
@@ -1,91 +1,36 @@
 {
-  "name": "@metamask/controllers",
-  "version": "29.0.1",
+  "name": "@metamask/controllers-monorepo",
+  "version": "0.0.0",
+  "private": true,
   "description": "Collection of platform-agnostic modules for creating secure data models for cryptocurrency wallets",
-  "keywords": [
-    "MetaMask",
-    "Ethereum"
-  ],
-  "homepage": "https://github.com/MetaMask/controllers#readme",
-  "bugs": {
-    "url": "https://github.com/MetaMask/controllers/issues"
-  },
-  "repository": {
-    "type": "git",
-    "url": "https://github.com/MetaMask/controllers.git"
-  },
-  "license": "MIT",
-  "main": "./dist/index.js",
-  "types": "./dist/index.d.ts",
-  "files": [
-    "dist/"
+  "workspaces": [
+    "packages/*"
   ],
   "scripts": {
-    "build": "rimraf dist && tsc --project tsconfig.build.json",
-    "build:link": "yarn build && cd dist && yarn link && rm -rf node_modules && cd ..",
+    "// 1": "'pre' commands are not supported by Yarn >= 2",
```

Comment:
See: https://yarnpkg.com/getting-started/migration#explicitly-call-the-pre-and-post-scripts

I'm not sure what the `prepublish` script was for — were we using this in an Action?

Reply:
Ah, not so for `prepublish`: https://yarnpkg.com/advanced/lifecycle-scripts

Maybe that was just used for manual publishing? Either it should be refactored to abide by their recs:
> the prepublish hook shouldn't have side effects

Reply:
Yes, it was for manual publishing. It was meant to ensure the project could not be published without building first. We should have been using `prepublishOnly`, as this was never meant to run on install.

For yarn v3, I don't really understand why `prepublishOnly` isn't documented as supported but migrating it to `yarn prepack` should work.

Reply:
Okay, gotcha. I'll make this change.

Reply:
This is effectively done because I'm now basing this branch off of #846.